### ISSUE TITLE
Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by chylvina** [cubiq/iscroll#402]
iscroll.js line 1100, Uncaught TypeError: Cannot read property 'length' of undefined.

<code>l = this.pages[i].length;</code>

**Comment by irothenbaum** [comment]
I'm getting this error with `snap : true` as well... Is there some unmentioned assumption about a required style or something I'm missing? Has anyone found a resolution to this?

**Comment by richgilbank** [comment]
@irothenbaum https://github.com/cubiq/iscroll/pull/570 Seems to be doing the trick for most.

**Comment by irothenbaum** [comment]
Hey rich, just realized my issue was not disabling vertical (there were no vertical elements, so it's related to the same problem mentioned above).

I tried deleting my comment, but I guess I wasn't fast enough... Thanks.

**Comment by jasonfb** [comment]
I found that I got this error when "snap" as passed the wrong parameter. In particular, I tried passing "snap" a css selector (like div.slide) didn't work, and even passing "div" didn't seem to work either, but when I correctly structured my HTML and passed simply true to the snap parameter, it started working for me.

I would recommend that IScroll have some error checking to see if you instantiate it correctly and politely throw an error if it can detect obvious developer-error. Would make this tool much more friendly to work with.
